enterprise/api/v1/urls.py

```diff
@@ -7,16 +7,15 @@
 
 from enterprise.api.v1 import views
 
-# Pylint considers module level variable as "Constant" and expects them to be upper-cased
-# that is why we have disabled 'invalid-name' check for variable definition below.
 router = DefaultRouter()  # pylint: disable=invalid-name
 router.register("site", views.SiteViewSet, 'site')
 router.register("auth-user", views.UserViewSet, 'auth-user')
-router.register(
-    "enterprise-course-enrollment",
-    views.EnterpriseCourseEnrollmentViewSet,
-    'enterprise-course-enrollment'
-)
+router.register("enterprise-customer-catalog", views.EnterpriseCustomerCatalogViewSet, 'enterprise-customer-catalog')
+router.register("enterprise-catalogs", views.EnterpriseCustomerCatalogApiViewSet, 'enterprise-catalogs')
```

Comment:
@douglashall Hmm, yeah from the ticket and the discussions I thought this endpoint would be the "list" version of its corresponding more detailed `/enterprise/api/v1/enterprise-catalogs/{uuid}/` endpoint, which would filter things (i.e. for details).

In terms of implementation, this would be a matter of changing the `list` function at https://github.com/edx/edx-enterprise/blob/master/enterprise/api/v1/views.py#L347 -- instead of querying the catalog client, we'd filter `EnterpriseCustomerCatalog` instances which the requesting user has access to, serialize them, and return the response.

I can do that in the Programs ticket but it's getting bigger and bigger and it'll slow me down a bit. *If* we have capacity for someone else to pick it up this sprint that'd be great, otherwise I'll fit it in as a separate commit, yeah. I've already refactored the Programs endpoint to point to its own dedicated view set, and that was significant in and of itself, so it might make sense to allocate this change to me as well.